src/index.js

```diff
@@ -33,3 +64,26 @@ if ( document.readyState !== 'loading' ) {
   document.addEventListener( 'DOMContentLoaded', onReady );
 }
 
+/* 
+REFLECTION
+
+Challenges: 
+	* Knowing what is already in the DOM and what is being added. in renderTheUI(), my instinct 
```

Comment:
There is definitely a degree of "developer ergonomics" here, but I think there's more to it than that. Are you familiar with the virtual DOM in react? (ng2 has a similar concept)